fuel-core/src/executor.rs

```diff
@@ -44,6 +48,32 @@ pub struct Executor {
 }
 
 impl Executor {
+    pub async fn submit_txs(&self, txs: Vec<Arc<Transaction>>) -> Result<(), Error> {
+        let db = self.database.clone();
+
+        for tx in txs.iter() {
+            // set status to submitted
+            db.update_tx_status(&tx.id(), TransactionStatus::Submitted { time: Utc::now() })?;
+        }
+
+        // setup and execute block
+        let current_height = db.get_block_height()?.unwrap_or_default();
+        let current_hash = db.get_block_id(current_height)?.unwrap_or_default();
```

Comment:
unwraping these might be source of inconsistencies. Maybe fail the function instead? both height and block id are critical for the correctness of the system

Reply:
Since the return type is `Result<Option<T>>`, errors are properly handled here via `?`, while `unwrap_or_default` is used on the option to initialize the block number if it hasn't been set in the db yet.